v2/tests/git_initializer_test.go

```diff
@@ -179,46 +235,201 @@ func TestMain(t *testing.T) {
-					if ok {
-						log.Println(logEntry.Message)
-					} else {
-						logEntryCh = nil
-					}
-				case err, ok := <-errCh:
-					if ok {
-						t.Fatalf("error from log stream: %s", err)
-					}
-					errCh = nil
-				case <-ctx.Done():
-					break
-				}
+			assertWorkerTerminalPhase(t, ctx, client, e, tc)
+			assertWorkerLogs(t, ctx, client, e, tc)
 
-				// log and err channels empty, let's check worker status
-				if logEntryCh == nil && errCh == nil {
-					for {
-						event, err := client.Core().Events().Get(ctx, e.ID)
-						require.NoError(t, err, "error getting event")
-
-						phase := event.Worker.Status.Phase
-						if phase.IsTerminal() {
-							require.Equal(t, tc.terminalPhase, phase, "worker's terminal phase does not match expected")
-							break
-						}
-					}
-					break
-				}
+			// We expect the vcs init container to fail for this test,
+			// therefore, assertions on jobs are not applicable
+			if tc.name == "GitHub - vcs failure" {
+				return
 			}
+
+			assertJobTerminalPhase(t, ctx, client, e, tc)
+			assertJobLogs(t, ctx, client, e, tc)
 		})
 	}
 }
+
+func assertWorkerTerminalPhase(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase,
+) {
+	statusCh, errCh, err := client.Core().Events().Workers().WatchStatus(
+		ctx,
+		e.ID,
+	)
+	require.NoError(
+		t,
+		err,
+		"error encountered attempting to watch worker status",
+	)
+
+	timer := time.NewTimer(testTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case status := <-statusCh:
+			phase := status.Phase
+			if phase.IsTerminal() {
+				require.Equal(
+					t,
+					tc.expectedWorkerPhase,
+					phase,
+					"worker's terminal phase does not match expected",
+				)
+				return
+			}
+		case err := <-errCh:
+			t.Fatalf("error encountered watching worker status: %s", err)
+		case <-timer.C:
+			t.Fatal("timeout waiting for worker to reach a terminal phase")
+		}
+	}
+}
+
+func assertWorkerLogs(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase,
+) {
+	// Check vcs container logs
+	assertLogs(
+		t,
+		ctx,
+		client,
+		e,
+		&core.LogsSelector{Container: "vcs"},
+		tc.expectedVCSLogsContains,
+	)
+
+	// We expect the vcs init container to fail for this test,
+	// therefore, assertions on worker logs are not applicable
+	if tc.name == "GitHub - vcs failure" {
+		return
+	}
+
+	// Check worker container logs
+	var contains string
+	if tc.expectedWorkerLogsContains != "" {
+		contains = tc.expectedWorkerLogsContains
+	} else {
+		// Look for default logs that we expect must exist
+		contains = "brigade-worker version"
+	}
+	assertLogs(
+		t,
+		ctx,
+		client,
+		e,
+		&core.LogsSelector{},
+		contains,
+	)
+}
+
+func assertJobTerminalPhase(
+	t *testing.T,
+	ctx context.Context,
+	client sdk.APIClient,
+	e core.Event,
+	tc testcase,
+) {
+	statusCh, errCh, err := client.Core().Events().Workers().Jobs().WatchStatus(
+		ctx,
+		e.ID,
+		testJobName,
```

Comment:
I'd still rather see job name passed in so that not all test cases are constrained to having only one job with one specific name. Similar for the `assertJobLogs` func.

After this, this PR is spectacular.

Reply:
Oops!  Now added.